File: Example_Objective-C/Example_Objective-C/Views/ShadowView.m
//
//  ShadowView.m
//  Example_Objective-C
//
//  Created by Babchenko Alexander on 8/22/19.
//  Copyright © 2019 UserReport. All rights reserved.
//

#import "ShadowView.h"

@implementation ShadowView

- (UIColor *)shadowColor {
    return [UIColor colorWithCGColor:self.layer.shadowColor];
}
- (void)setShadowColor:(UIColor *)shadowColor {
    self.layer.shadowColor = [shadowColor CGColor];
}

- (CGFloat)shadowOpacity {
    return self.layer.shadowOpacity;
}
- (void)setShadowOpacity:(CGFloat)shadowOpacity {
    self.layer.shadowOpacity = shadowOpacity;
}

- (CGPoint)shadowOffset {
    return CGPointMake(self.layer.shadowOffset.width, self.layer.shadowOffset.height);
}
- (void)setShadowOffset:(CGPoint)shadowOffset {
    self.layer.shadowOffset = CGSizeMake(shadowOffset.x, shadowOffset.y);
}

- (CGFloat)shadowRadius {
    return self.layer.shadowRadius;
}
- (void)setShadowRadius:(CGFloat)shadowRadius {
    self.layer.shadowRadius = shadowRadius;
}

- (CGFloat)cornerRadius {
    return self.layer.cornerRadius;
}
- (void)setCornerRadius:(CGFloat)cornerRadius {
    self.layer.cornerRadius = cornerRadius;
}


@end

File: Example/Example/Views/ShadowView.swift
//
//  Copyright © 2017 UserReport. All rights reserved.
//

import UIKit

@IBDesignable class ShadowView: UIView {
    
    @IBInspectable var shadowColor: UIColor? {
        set { layer.shadowColor = newValue!.cgColor }
        get { return layer.shadowColor?.uiColor }
    }

    @IBInspectable var shadowOpacity: Float {
        set { layer.shadowOpacity = newValue }
        get { return layer.shadowOpacity }
    }
    
    @IBInspectable var shadowOffset: CGPoint {
        set { layer.shadowOffset = CGSize(width: newValue.x, height: newValue.y) }
        get { return CGPoint(x: layer.shadowOffset.width, y:layer.shadowOffset.height) }
    }
    
    @IBInspectable var shadowRadius: CGFloat {
        set { layer.shadowRadius = newValue }
        get { return layer.shadowRadius }
    }
    
    @IBInspectable var cornerRadius: CGFloat {
        set { layer.cornerRadius = newValue }
        get { return layer.cornerRadius  }
    }
}

fileprivate extension CGColor {
    var uiColor: UIKit.UIColor {
        return UIColor(cgColor: self)
    }
}
